src/main/kotlin/com/sourcegraph/cody/context/CurrentlyOpenFileListener.kt

```diff
@@ -11,15 +11,18 @@ class CurrentlyOpenFileListener(
     private val embeddingStatusView: EmbeddingStatusView
 ) : FileEditorManagerListener {
   override fun selectionChanged(event: FileEditorManagerEvent) {
-    ApplicationManager.getApplication().runReadAction {
-      val newFile = event.newFile
-      val openedFileName = newFile?.name ?: ""
-      var relativeFilePath: String? = null
+    val newFile = event.newFile
+    val openedFileName = newFile?.name ?: ""
+    var relativeFilePath: String? = null
 
+    ApplicationManager.getApplication().executeOnPooledThread {
       if (newFile != null) {
         relativeFilePath = ProjectFileUtils.getRelativePathToProjectRoot(project, newFile)
       }
-      embeddingStatusView.setOpenedFileName(openedFileName, relativeFilePath)
+
+      ApplicationManager.getApplication().invokeLater {
```

Comment:
ui actions should be called from ui thread

Reply:
Cannot that whole code be simplified to just that?

```
ApplicationManager.getApplication().executeOnPooledThread {
  if (newFile?.name) {
    val relativeFilePath = ProjectFileUtils.getRelativePathToProjectRoot(project, newFile)
    ApplicationManager.getApplication().invokeLater {
      embeddingStatusView.setOpenedFileName(newFile.name, relativeFilePath)
    }
  }
}
```